lms/djangoapps/certificates/api.py

```diff
@@ -425,51 +396,66 @@ def example_certificates_status(course_key):
     return ExampleCertificateSet.latest_status(course_key)
 
 
+def _safe_course_key(course_key):
+    if not isinstance(course_key, CourseKey):
+        try:
+            return CourseKey.from_string(course_key)
+        except InvalidKeyError:
+            log.warning('Unable to parse course_key "%s"', course_key, exc_info=True)
+            return
```

Comment:
I see a couple of issues with this:

1. It is typically not best practice to mask exceptions and return `None` in its stead.  This eventually leads to long-hours of debugging when things go faulty.  By letting the exception move forward, you will get a better stack trace in error situations.
2. Is it necessary to have the same variable support multiple types for 
`course_key` (both `string` and `CourseKey`)?  We typically do not do this since we want the interface to be clear.  Python [duck-typing](https://en.wikipedia.org/wiki/Duck_typing) does allow/encourage passing in different types for the same object - however, all types need to support the same required interface - and calling `isinstance` is frowned upon.
3. In searching the codebase for callers, it looks like all of them pass in a `CourseKey` object.  (Maybe in the past all of them didn't.)

Reply:
Yeah, I was trying to keep the functionality of the previous code mostly intact while making these changes, and that code was pretty defensive about course keys being either strings or CourseKey objects, so I erred more on the "clean things up for the sake of code readability".  I can pluck out some of the exception-masking, though.